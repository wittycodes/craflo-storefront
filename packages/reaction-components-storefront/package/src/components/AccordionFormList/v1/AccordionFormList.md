### Overview

The `AccordionFormList` component renders an `Accordion` where each row can be expanded to reveal an edit form for that row's data, and where the final row allows you to add a new item to the list.

### Usage

```jsx
const items = [
  { id: "1", detail: "Detail", label: "Label" },
  { id: "2", detail: "Detail", label: "Label" },
  { id: "3", detail: "Detail", label: "Label" }
];

const handleItemDeleted = id => {
  console.log("Deleting", id);
};

class MyAddForm extends React.Component {
  submit() {
    // AccordionFormList will call this when the submit button is clicked
    console.log("Adding");
  }

  render() {
    return (
      <div
        style={{ textAlign: "center", fontWeight: 700, fontStyle: "italic" }}
      >
        There would be an add form here
      </div>
    );
  }
}

class MyEditForm extends React.Component {
  submit() {
    // AccordionFormList will call this when the submit button is clicked
    console.log("Editing");
  }

  render() {
    return (
      <div
        style={{ textAlign: "center", fontWeight: 700, fontStyle: "italic" }}
      >
        There would be an edit form here
      </div>
    );
  }
}

<AccordionFormList
  components={{
    ItemAddForm: MyAddForm,
    ItemEditForm: MyEditForm
  }}
  items={items}
  onItemDeleted={handleItemDeleted}
/>;
```

### Theme

Assume that any theme prop that does not begin with "rui" is within `rui_components`. See [Theming Components](./#!/Theming%20Components).

| Theme Prop                                          | Default          | Description                                                                                         |
| --------------------------------------------------- | ---------------- | --------------------------------------------------------------------------------------------------- |
| `Accordion.borderColor`                             | `black10`        | Border color for the Accordion and the "Add an item" area below it                                  |
| `Accordion.borderStyle`                             | `solid`          | Border style for the Accordion and the "Add an item" area below it                                  |
| `Accordion.borderWidth`                             | `1px`            | Border width for the Accordion and the "Add an item" area below it                                  |
| `Accordion.borderRadius`                            | `2px`            | Border radius for the Accordion and the "Add an item" area below it                                 |
| `AccordionFormList.addActionPaddingBottom`          | `14px`           | Bottom padding for the "Add an item" button area below the accordion                                |
| `AccordionFormList.addActionPaddingLeft`            | `16px`           | Left padding for the "Add an item" button area below the accordion                                  |
| `AccordionFormList.addActionPaddingRight`           | `16px`           | Right padding for the "Add an item" button area below the accordion                                 |
| `AccordionFormList.addActionPaddingTop`             | `14px`           | Top padding for the "Add an item" button area below the accordion                                   |
| `AccordionFormList.actionButtonColor`               | `coolGreyActive` | Color for the "Add an item" button when not hovered                                                 |
| `AccordionFormList.actionButtonHoverColor`          | `coolGreyHover`  | Color for the "Add an item" button when hovered                                                     |
| `AccordionFormList.actionButtonIconColor`           | `coolGrey500`    | "Add an item" icon color                                                                            |
| `AccordionFormList.actionButtonIconHeight`          | `20px`           | "Add an item" icon height                                                                           |
| `AccordionFormList.actionButtonIconMarginRight`     | `10px`           | Spacing between the icon and "Add an item" text                                                     |
| `AccordionFormList.actionButtonIconWidth`           | `20px`           | "Add an item" icon width                                                                            |
| `AccordionFormList.actionPaddingBottom`             | `16px`           | Bottom padding for the area within the accordion item where delete, save, and cancel buttons appear |
| `AccordionFormList.actionPaddingLeft`               | `0`              | Bottom padding for the area within the accordion item where delete, save, and cancel buttons appear |
| `AccordionFormList.actionPaddingRight`              | `0`              | Bottom padding for the area within the accordion item where delete, save, and cancel buttons appear |
| `AccordionFormList.actionPaddingTop`                | `16px`           | Bottom padding for the area within the accordion item where delete, save, and cancel buttons appear |
| `AccordionFormList.spaceBetweenActiveActionButtons` | `16px`           | Space between active action buttons                                                                 |
| `AccordionFormList.actionDeleteButtonHoverColor`    | `redHover`       | Delete button color when hovered                                                                    |
