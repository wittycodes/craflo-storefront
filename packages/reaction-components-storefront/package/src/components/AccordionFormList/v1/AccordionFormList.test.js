import React from "react";
import renderer from "react-test-renderer";
import mockComponents from "../../../tests/mockComponents";
import AccordionFormList from "./AccordionFormList";

test("basic snapshot", () => {
  const component = renderer.create(<AccordionFormList components={mockComponents} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
