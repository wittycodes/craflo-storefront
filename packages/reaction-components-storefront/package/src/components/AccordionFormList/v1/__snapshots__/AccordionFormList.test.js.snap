// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
.c0 {
  border-color: #e6e6e6;
  border-style: solid;
  border-width: 1px;
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
  border-top-width: 1px;
  box-sizing: border-box;
  color: inherit;
  overflow: hidden;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  color: #505558;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 400;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
  color: #3b5c70;
  cursor: pointer;
  display: table;
}

.c1:hover {
  color: #5d8fa9;
}

.c1:hover svg {
  color: inherit !important;
}

.c2 {
  color: inherit;
  height: 20px;
  margin: 0;
  margin-right: 10px;
  width: 20px;
}

.c2 svg {
  color: #3c3c3c;
  fill: currentColor;
  height: 1em;
  width: 1em;
  vertical-align: middle;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
      tabIndex={0}
    >
      <span
        className="c2"
      >
        iconPlus
      </span>
      Add an item
    </div>
  </div>
</div>
`;
